### DISCUSSION TITLE
How to manually update the WebView2 Runtime?

### DISCUSSION THREAD

**Discussion by ajtruckle** [MicrosoftEdge/WebView2Feedback#4036] [Q&A]
User is running Windows 10, 64 bit.

My application won't run and shuts down.

They are using:

- WebView2 Runtime Version: 108.0.1462.54
- WebView2 SDK Build: 1661.34

I am using:

- WebView2 Runtime Version: 111.0.1661.62
- WebView2 SDK Build: 1661.34

There is quite a difference between `108.0.1462.54` and `111.0.1661.62`. How can I get it up to date?

**Comment by ajtruckle** [comment]
Think I found the solution on the internet:

To install or update the WebView2 Runtime:

1. Go to page https://developer.microsoft.com/en-us/microsoft-edge/webview2/.
2. Scroll down to section **Evergreen Bootstrapper**, select **Download** > **Accept and Download**.
3. Find the executable file in your **Downloads** folder, right-click it and select **Run as administrator**.
4. Enter your administrator credentials. The installation starts and is completed in the background.

After I ran this my runtime went from `111.0.1661.62` to `112.0.1722.34`. I am now checking with the user in question.

---

By the way, according to this [link](https://learn.microsoft.com/en-us/deployoffice/webview2-install) it does say:

> To keep WebView2 up to date with the latest security and reliability updates, WebView2 Runtime _uses its own automatic update process, separate from the update channel used by Microsoft 365 Apps_. WebView2 Runtime is updated approximately a few times each month. Updates usually range in size from 5 MB to 30 MB, depending on the type of update. Differential updates and Delivery Optimization are used where possible. For information about managing these updates, see [Managing WebView2 applications](https://learn.microsoft.com/en-us/microsoft-edge/webview2/concepts/enterprise).

**Comment by ajtruckle** [comment]
The user has now managed to update to the latest runtime and my software runs again. This begs the question:

_Why was his runtime not up to date if the operating system is supposed to manage it?_ 🤔

**Comment by sln162** [comment]
> According to my current test, WebView2 is installed on the Win10 machine through Edge's task plan/service (the actual test should be task plan)    #3317  

I guess that the relevant task plan may not be executed, or EdgeUpdate is not installed or the file is missing

**Comment by novac42** [comment]
@ajtruckle Can you collect the installation log?

**Comment by ajtruckle** [comment]
@novac42 We have asked this before. There is no simple way to obtain a log.

**Comment by LiangTheDev** [comment]
WebView2 Runtime is designed to auto update, using scheduled tasks. Auto update would not work if the IT admin setup policy to block it, those scheduled tasks are disabled. We need [Edge update/installer logs](https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/diagnostics/install.md#installer-logs) to tell us what's going on.

**Reply by bananalemonmelon** [reply]
Auto update only? Seriously no manual way? 
You don't think about people that constantly in metered-connection mode that need to manually control the update? You think everyone is in first-world country and can afford unlimited data?

**Comment by ghost** [comment]
I found a way to manually update it however not an elegant one.

Go to `Control Panel > Programs > Programs and Features`, right click on `Microsoft Edge WebView2 Runtime` and select `Change`. In a window that opens in a window that opens confirm that you want to `Restore`. That automatically downloaded and installed the latest version for me.

**Reply by chaixuqing** [reply]
Thank you very much. This successfully solved my problem.
